soc/nordic/common/mram_latency.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * nRF SoC specific public APIs for MRAM latency management
+ */
+
+#ifndef SOC_NORDIC_COMMON_MRAM_LATENCY_H_
+#define SOC_NORDIC_COMMON_MRAM_LATENCY_H_
+
+#include <zephyr/sys/onoff.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** @internal For test purposes only. */
+extern struct onoff_manager mram_latency_mgr;
+
+/** @brief Request MRAM operations without latency.
+ *
+ * The return value indicates the success or failure of an attempt to initiate
+ * an operation to request the MRAM low latency. If initiation of the
+ * operation succeeds, the result of the request operation is provided through
+ * the configured client notification method, possibly before this call returns.
+ *
+ * @param cli pointer to client state providing instructions on synchronous
+ *            expectations and how to notify the client when the request
+ *            completes. Behavior is undefined if client passes a pointer
+ *            object associated with an incomplete service operation.
+ *
+ * @retval non-negative the observed state of the on-off service associated
+ *                      with the MRAM latency service.
+ * @retval -EIO if MRAM latency service returned error.
+ * @retval -EINVAL if the parameters are invalid.
+ * @retval -EAGAIN if the reference count would overflow.
+ */
+int mram_no_latency_request(struct onoff_client *cli);
```

Comment:
an additional sync API may be useful for simplicity, dealing with onoff is cumbersome

Reply:
added `mram_no_latency_sync_request()` and `mram_no_latency_sync_release()`